### PR TITLE
StackPulse Integration

### PR DESCRIPTION
**Pull request by hagaishapira**
### What does this PR do?

Added a new integration for StackPulse based in the webhook integration and a StackPulse step that posts events back to Datadog.

### Motivation

Enabling users to easily configure triggers from Datadog to run StackPulse playbooks.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: stackpulse/manifest.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "display_name": "StackPulse",
+  "maintainer": "support@stackpulse.io",
+  "manifest_version": "1.0.0",
+  "name": "stackpulse",
+  "metric_prefix": "stackpulse.",
```
Comment (hithwen) [inline stackpulse/manifest.json:6]:
```suggestion
  "metric_prefix": "",
```
No need to se it since there are no metrics on this integration

**Review by ChristineTChen — APPROVED** [review]
LGTM!

File: stackpulse/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# StackPulse Integration
+
+## Overview
+
+The [StackPulse][1] integration enables you to trigger automatic playbooks in response to Datadog alerts, providing alert enrichment, incident mitigation and collaboration. You can then send back events from your playbook executions directly to your Datadog event stream and dedicated StackPulse dashboard.
```
Comment (apigirl) [inline stackpulse/README.md:5]:
```suggestion
The [StackPulse][1] integration enables you to trigger automatic playbooks in response to Datadog alerts, providing alert enrichment, incident mitigation, and collaboration. You can then send back events from your playbook executions directly to your Datadog event stream and dedicated StackPulse dashboard.
```

File: stackpulse/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+## Setup
+
+To set up this integration you must have an active [StackPulse account][2] and an Account Owner role in that account. You must also have proper admin permissions in Datadog.
```
Comment (apigirl) [inline stackpulse/README.md:9]:
```suggestion
To set up this integration, you must have an active [StackPulse account][2] and an Account Owner role in that account. You must also have proper admin permissions in Datadog.
```

File: stackpulse/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+### In StackPulse
+
+1. On the **Integrations** page under **Monitoring**, locate the **Datadog** card and click **New**.
```
Comment (apigirl) [inline stackpulse/README.md:13]: might be good to directly link to this page is possible

File: stackpulse/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    ![Datadog Card][3]
+
+2. Provide a meaningful name for the integration and click **Add**.
+
+3. **Copy** the newly created webhook endpoint.
+
+### In Datadog
+
+1. Navigate to **Integrations** and choose the **Webhooks** card.
```
Comment (apigirl) [inline stackpulse/README.md:23]: Adding a link here would be good too.

File: stackpulse/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    ![Datadog Webhook Card][4]
+
+2. Click on **New** to add a new Webhook integration.
+
+3. Fill in the name for a new Webhook integration (remember to use it later in specific DataDog Monitors to trigger StackPulse) and the Webhook URL from the previous step.
```
Comment (apigirl) [inline stackpulse/README.md:29]:
```suggestion
3. Fill in the name for a new Webhook integration (remember to use it later in specific Datadog monitors to trigger StackPulse) and the Webhook URL from the previous step.
```

File: stackpulse/README.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+4. It is recommended to enhance the payload with additional alert information, using portions of a configuration below:
```
Comment (apigirl) [inline stackpulse/README.md:31]:
```suggestion
4. StackPulse recommends enhancing the payload with additional alert information, using portions of a configuration below:
```

File: stackpulse/README.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            "TAGS": "$TAGS",
+            "TEXT_ONLY_MSG": "$TEXT_ONLY_MSG",
+            "USER": "$USER",
+            "USERNAME": "$USERNAME",
+            "LOGS_SAMPLE": "$LOGS_SAMPLE"
+        }
+    }
+    ```
+
+5. Pick Monitors for triggering StackPulse Playbooks and in the **Alert Your Team** field add a reference to the newly created Webhook integration. For further details, please refer to [DataDog Documentation on Managing Monitors](https://docs.datadoghq.com/monitors/manage_monitor/).
```
Comment (apigirl) [inline stackpulse/README.md:72]:
```suggestion
5. Pick monitors for triggering StackPulse Playbooks, and in the **Alert Your Team** field, add a reference to the newly created Webhook integration. For further details, see the [Datadog documentation on managing monitors](https://docs.datadoghq.com/monitors/manage_monitor/).
```

File: stackpulse/README.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+6. Navigate to **Integrations -> APIs** and choose the **API Keys** card. Under **New API Key** input a meaningful name for the new key and click **Create API Key**, then **Copy** the new masked key.
```
Comment (apigirl) [inline stackpulse/README.md:74]:
```suggestion
6. Navigate to **Integrations -> APIs** and choose the **API Keys** card. Under **New API Key**, input a meaningful name for the new key and click **Create API Key**, then **Copy** the new key.
```

File: stackpulse/README.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+## Data Collected
+
+### Metrics
+
+The StackPulse integration does not provide any metrics.
+
+### Events
+
+The StackPulse integration allows you to send events to you Datadog Event Stream using the [Datadog Post Event][6] step. You can use the step with your playbooks to notify Datadog about successful mitigations, execution failures, etc.
```
Comment (apigirl) [inline stackpulse/README.md:92]:
```suggestion
The StackPulse integration allows you to send events to your Datadog event stream using the [Datadog Post Event][6] step. You can use the step with your playbooks to notify Datadog about successful mitigations and execution failures.
```

**Review by apigirl — COMMENTED** [review]
some docs feedback

**Comment by hagaishapira** [comment]
Hi @apigirl, thanks for the great comments. I've fixed them all so let me know if there is anything else I should chance :pray:

**Pull request merged**